zipline/sources/test_source.py

```diff
@@ -153,10 +150,21 @@ def __init__(self, *args, **kwargs):
                 self.event_list[1].dt - self.event_list[0].dt)
             self.concurrent = kwargs.get('concurrent', False)
 
+            self.identifiers = kwargs.get(
+                'sids',
+                np.unique([event.sid for event in self.event_list]).tolist())
+            env.update_asset_finder(erase_existing=True,
```

Comment:
I'm not excited about constantly wiping the asset finder like this.  I'd rather just construct a new one and set it.